debezium-server/debezium-server-redis/src/main/java/io/debezium/server/redis/RedisStreamChangeConsumer.java

```diff
@@ -118,62 +103,106 @@ void close() {
         catch (Exception e) {
             LOGGER.warn("Exception while closing Jedis: {}", client, e);
         }
+        finally {
+            client = null;
+        }
+    }
+
+    /**
+    * Split collection to batches by batch size using a stream
+    */
+    private <T> Stream<List<T>> batches(List<T> source, int length) {
+        if (length <= 0) {
+            throw new IllegalArgumentException("length = " + length);
+        }
+
+        int size = source.size();
+        if (size <= 0) {
+            return Stream.empty();
+        }
+
+        int fullChunks = (size - 1) / length;
+        return IntStream.range(0, fullChunks + 1).mapToObj(
+                n -> source.subList(n * length, n == fullChunks ? size : (n + 1) * length));
     }
 
     @Override
     public void handleBatch(List<ChangeEvent<Object, Object>> records,
                             RecordCommitter<ChangeEvent<Object, Object>> committer)
             throws InterruptedException {
+        DelayStrategy delayStrategy = DelayStrategy.exponential(initialRetryDelay, maxRetryDelay);
```

Comment:
How comes we create this for each invocation?

Reply:
I saw that once it's created in the postconstruct as it was, the delay is *always* 300ms (the initialRetryDelay). I have not delved into this, but moved it to be created in every batch handle. Maybe I missed something here, what do you think?

Reply:
Hum, that's odd, I don't see why it shouldn't work when set up once in post construct. But I don't think it matters that much, should be fine to set this up once per batch.